### PR TITLE
feature: added possibility to define the name of the instance and backup storage class

### PR DESCRIPTION
**Pull request by leonsteinhaeuser**
As part of our customization's to the PostgreSQL cluster, we discovered that it is currently not possible to change the storage class for predefined instances. Since our cluster infrastructure contains more than one storage class with different underlying storage technologies, it is necessary for us to select a specific storage class. We found that this is already possible when we define our own instance, but we thought it would be more useful to simply add this functionality to the chart.

### CODE REVIEW

File: helm/postgres/templates/postgres.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ spec:
     - name: {{ default "instance1" .Values.instanceName | quote }}
       replicas: {{ default 1 .Values.instanceReplicas }}
       dataVolumeClaimSpec:
+        {{- if .Values.storageClassName -}}storageClassName: {{ .Values.storageClassName | quote }}{{ end }}
```
Comment (benjaminjb) [inline helm/postgres/templates/postgres.yaml:24]:
```suggestion
        {{- if .Values.instanceStorageClassName }}
        storageClassName: {{ .Values.instanceStorageClassName | quote }}
        {{- end }}
```
A) changing the spacing here and below to keep with the pattern already established.
B) `values.yaml` has a `instanceStorageClassName` field, not. `storageClassName` field

File: helm/postgres/templates/postgres.yaml [hunk 2]
```diff
@@ -56,6 +57,7 @@ spec:
         {{- if $repo.volume }}
         volume:
           volumeClaimSpec:
+            {{- if $repo.volume.backupsStorageClassName -}}storageClassName: {{ .Values.backupsStorageClassName | quote }}{{ end }}
```
Comment (benjaminjb) [inline helm/postgres/templates/postgres.yaml:60]:
```suggestion
            {{- if $repo.volume.backupsStorageClassName }}
            storageClassName: {{ .Values.backupsStorageClassName | quote }}
            {{- end }}
```

File: helm/postgres/templates/postgres.yaml [hunk 3]
```diff
@@ -114,6 +116,7 @@ spec:
       - name: repo1
         volume:
           volumeClaimSpec:
+            {{- if .Values.backupsStorageClassName -}}storageClassName: {{ .Values.backupsStorageClassName | quote }}{{ end }}
```
Comment (benjaminjb) [inline helm/postgres/templates/postgres.yaml:119]:
```suggestion
            {{- if .Values.backupsStorageClassName }}
            storageClassName: {{ .Values.backupsStorageClassName | quote }}
            {{- end }}
```

**Comment by benjaminjb** [comment]
Hello, thanks for the PR, I've made a few comments on it to (a) correct a slight mistake and (b) bring it in line with the multiline `if...end` format we've established.

Also, at least for the backup `repo` section, have you ever tried to put all the necessary info into the `pgBackRestConfig` field in the `values.yaml`? I wonder how that would work for that specific part of the template.

**Comment by leonsteinhaeuser** [comment]
@benjaminjb thank you very much for your reply and review. I will look at it asap and update the PR.

**Review by ValClarkson — APPROVED** [review]

**Pull request merged**